ctapipe_io_lst/__init__.py

```diff
@@ -341,7 +346,13 @@ def _generator(self):
 
             # gain select and calibrate to pe
             if self.r0_r1_calibrator.calibration_path is not None:
-                self.r0_r1_calibrator.calibrate(array_event)
+
+                # skip flatfield and pedestal events if asked
+                if (not (array_event.trigger.event_type == EventType.FLATFIELD or
+                    array_event.trigger.event_type == EventType.SKY_PEDESTAL) or
+                         self.calibrate_flatfields_and_pedestals):
```

Comment:
This might be more readable:

```
if array_event.trigger.event_type not in {EventType.FLATFIELD, EventType.SKY_PEDESTAL} or self.calibrate_flatfields_and_pedestals
```

Reply:
:-) I knew you had a better way ;-)